File: Entities/Concrete/Product.cs
using Core.Entities;
using Entities.Enums;
using System.Collections.Generic;

namespace Entities.Concrete
{
    public class Product : ProjectBaseEntity, IEntity
    {
        public string Name { get; set; }
        public Size Size  { get; set; }//boyut (S,M,L,XL,XXL..)
        public int ColorId { get; set; }
        public virtual Color Color { get; set; }
        public virtual List<WareHouseProductMapping> WareHouseProductMappings { get; set; }
        public virtual List<Order> Orders { get; set;}

    }
}

File: UIPreparation/UI/src/app/core/components/app/product/models/product.ts
import { Order } from "../../order/models/Order";
import { WareHouseProductMapping } from "../../wareHouseProductMapping/models/WareHouseProductMapping";

export class Product{
    id?:number; 
createdUserId?:number; 
createdDate?:(Date | any); 
lastUpdatedUserId?:number; 
lastUpdatedDate?:(Date | any); 
status:boolean; 
isDeleted:boolean; 
name?:string; 
colorId?:number; 
wareHouseProductMappings?:WareHouseProductMapping[]; 
orders?:Order[]; 

}